salt/files/master.d/f_defaults.conf

```diff
@@ -697,6 +695,14 @@ ext_pillar:
 # master config file that can then be used on minions.
 {{ get_config('pillar_opts', 'True') }}
 
+# The pillar_source_merging_strategy option allows you to configure merging strategy
+# between different sources. It accepts four values: recurse, aggregate, overwrite,
+# or smart. Recurse will merge recursively mapping of data. Aggregate instructs
+# aggregation of elements between sources that use the #!yamlex renderer. Overwrite
+# will verwrite elements according the order in which they are processed. This is
+# behavior of the 2014.1 branch and earlier. Smart guesses the best strategy based
+# on the "renderer" setting and is the default value.
+{{ get_config('pillar_source_merging_strategy', 'smart') }}
```

Comment:
You're right. Looks like it was added in the last couple of days by someone else. The change to the ACL is still worthwhile though.

My pillar data looks like this:-

```yaml
salt:
  master:
    client_acl:
      johncc:
        - 'cdr*':
          - .*
        - 'beta*':
          - .*
        - 'cmstest*':
          - .*
        - test.*
      danc:
        - 'cdr*':
          - .*
        - 'beta*':
          - .*
        - 'cmstest*':
          - .*
        - test.*
```

The old code would not cope with this.